tuna/miopen/miopen_tables.py

```diff
@@ -530,44 +525,48 @@ class BNGolden(BASE, GoldenMixin):
   kernel_group = Column(Integer, nullable=True)
 
 
-class BenchmarkTable(BASE):
+class BenchmarkMixin():
+  """Mixin class for bechmark tables"""
+
+  @declared_attr
+  def framework(self):
+    """Framework Fkey"""
+    return Column(Integer, ForeignKey("framework.id"), nullable=False)
+
+  @declared_attr
+  def model(self):
+    """Model Fkey"""
+    return Column(Integer, ForeignKey("model.id"), nullable=False)
+
+  batchsize = Column(Integer, nullable=False, server_default="32")
+  gpu_number = Column(Integer, nullable=True, server_default="1")
+  driver_cmd = Column(String(length=512), nullable=False)
+
+
+class ConvolutionBenchmark(BASE, BenchmarkMixin):
   """benchmark table for framework and model parameters"""
-  __tablename__ = "benchmark"
+  __tablename__ = "conv_benchmark"
   __table_args__ = (UniqueConstraint("framework",
                                      "model",
                                      "batchsize",
                                      "gpu_number",
-                                     "driver_cmd",
+                                     "config",
                                      name="uq_idx"),)
 
-  framework = Column(Integer, ForeignKey("framework.id"), nullable=False)
-  model = Column(Integer, ForeignKey("model.id"), nullable=False)
-  batchsize = Column(Integer, nullable=False, server_default="32")
-  gpu_number = Column(Integer, nullable=True, server_default="1")
-  driver_cmd = Column(String(length=128), nullable=False)
+  config = Column(Integer, ForeignKey("conv_config.id"), nullable=False)
 
 
-class ConvBenchPerfTable(BASE):
-  """benchmark table for performance parameters"""
-  __tablename__ = "conv_benchmark_perf"
-  __table_args__ = (UniqueConstraint("config",
-                                     "benchmark",
-                                     "solver",
-                                     "rocm_v",
-                                     "miopen_v",
+class BNBenchmark(BASE, BenchmarkMixin):
+  """benchmark table for framework and model parameters"""
+  __tablename__ = "bn_benchmark"
+  __table_args__ = (UniqueConstraint("framework",
+                                     "model",
+                                     "batchsize",
+                                     "gpu_number",
```

Comment:
I think its because we can run benchmarks on machines with different number of GPUs and that might affect the overall benchmark performance. @JehandadKhan want to chime in?

Reply:
I think it would help if we rename it to `gpu_count` ? Since multi-GPU benchmarks have a number of GPUs associated with them. MIOpen still runs each config on a single GPU though

Reply:
OK, should we default it to 1 if not specified?